be/src/exec/pipeline/exchange/exchange_sink_operator.h

```diff
@@ -205,7 +209,7 @@ class ExchangeSinkOperatorFactory final : public OperatorFactory {
     void close(RuntimeState* state) override;
 
 private:
-    std::shared_ptr<SinkBuffer> _buffer;
+    std::shared_ptr<ExchangeSinkOperator::BufferType> _buffer;
```

Comment:
How about make SinkBuffer an abstract interface.
And your new added Buffer can be on implementation.
Then we can easily to switch between different implementation and see their differences.

Reply:
> How about make SinkBuffer an abstract interface. And your new added Buffer can be on implementation. Then we can easily to switch between different implementation and see their differences.

good idea~ I will try it